packages/components/src/components/Dropzone/Dropzone.tsx

```diff
@@ -0,0 +1,252 @@
+  let iconColor: IconProps["color"] = "colorIconInfo";
+  if ((!isDragActive && !isDragReject) || status === "success") {
+    icon = "CheckCircleIcon";
+    iconColor = "colorIconSuccess";
+  }
+  if (isDragAccept || status === "default" || status === "loading") {
+    icon = "CloudArrowUpIcon";
+    iconColor = "colorIconInfo";
+  }
+  if (isDragReject || status === "error" || status === "uploadError") {
+    icon = "ExclamationTriangleIcon";
+    iconColor = "colorIconError";
+  }
+
+  return <Icon color={iconColor} decorative icon={icon} size={"sizeIcon40"} />;
+};
+
+const Dropzone = React.forwardRef<HTMLDivElement, DropzoneProps>(
+  (
+    {
+      fileTypes = { "application/pdf": [".pdf"] },
+      cancelDocumentUpload,
+      sendDocument,
+    },
+    ref
+  ) => {
+    const [status, setStatus] = useState("default");
+    const [progress, setProgress] = useState(0);
+    const [errors, setErrors] = useState("");
+    const {
+      acceptedFiles,
+      getRootProps,
+      getInputProps,
+      isDragActive,
+      isDragAccept,
+      isDragReject,
+      isFocused,
+    } = useDropzone({
+      maxFiles: 1,
+      maxSize: MAX_FILE_SIZE,
+      multiple: false,
+      accept: {
+        ...fileTypes,
+      },
+      onDropAccepted: () => {
+        setStatus("loading");
+        sendDocument(acceptedFiles[0], (progress) => {
+          setProgress(progress);
+        })
+          .then(() => {
+            setProgress(0);
+            setStatus("success");
+          })
+          .catch(() => {
+            setStatus("uploadError");
+          });
+      },
+      onDropRejected: (fileRejections) => {
+        setStatus("error");
+        forEach(fileRejections, (file) => {
+          forEach(file.errors, (err) => {
+            if (err.code === "file-too-large") {
+              setErrors("File must be less than 50mb.");
+            }
+            if (err.code === "file-invalid-type") {
+              setErrors("Wrong file type. PDF format only.");
+            }
+          });
+        });
+      },
+    });
+
+    React.useEffect(() => {
+      if (isDragActive) {
+        setStatus("default");
+      }
+    }, [isDragActive]);
+
+    return (
+      <>
+        <Box.div
+          {...getVariants(status, isDragAccept, isDragReject, isFocused)}
+          alignItems="center"
+          borderRadius={"borderRadius30"}
+          borderStyle={"borderDashed"}
+          borderWidth={"borderWidth10"}
+          display="flex"
+          flexDirection="column"
+          padding={"space70"}
+          {...getRootProps({
+            status,
+            isDragActive,
+            isDragAccept,
+            isDragReject,
+          })}
+          ref={ref}
+        >
+          <input {...getInputProps()} />
+          {getIcon(status, isDragActive, isDragAccept, isDragReject)}
+
+          {!isDragActive && status === "loading" && (
+            <>
+              {acceptedFiles[0].name}
```

Comment:
What happens if there are multiple file types that need to be uploaded. This look like it checks just the first type.

Reply:
This is just the first file in the array. If there are more files we could map over them and call sendDocument on them?